lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,426 @@
+<template>
+	<div
+		class="ds-modal"
+		:class="{
+			'-danger': danger,
+		}"
+		@click.stop="$emit('close-modal')"
+	>
+		<div class="ds-modal__wrapper" :class="{ '-small': size === MODAL_SIZES.SMALL }">
+			<div
+				v-if="headerImage"
+				class="ds-modal__image"
+				:style="{ backgroundImage: 'url(' + headerImage + ')' }"
+			></div>
+			<div class="ds-modal__content" :class="{ '-centered': contentCentered }">
+				<div class="ds-modal__header">
+					<wnl-icon-button
+						touchable
+						:icon="ICONS.FA_XMARK"
+						class="ds-modal__close"
+						:size="ICON_SIZES.SMALL"
+						:elevation="BUTTON_ELEVATIONS.X_SMALL"
+						:color="ICON_BUTTON_COLORS.NEUTRAL_WEAK"
+						@click.native.stop="$emit('close-modal')"
+					/>
+					<feature-icon
+						v-if="headerFeatureIcon"
+						class="ds-modal__headerFeatureIcon"
+						:color="calcHeaderFeatureIconColor"
+						:icon="headerFeatureIcon"
+						:size="FEATURE_ICON_SIZES.X_LARGE"
+						double-background
+					/>
+					<h4
+						class="ds-modal__headerTitle"
+						:class="{ '-small': headerTitleSize === MODAL_HEADER_TITLE_SIZES.SMALL }"
+						>{{ headerTitle }}</h4
+					>
+					<h5 v-if="headerSubtitle" class="ds-modal__headerSubtitle">{{
+						headerSubtitle
+					}}</h5>
+				</div>
+				<div v-if="$slots.default" class="ds-modal__slotContent">
+					<slot />
+				</div>
+				<div class="ds-modal__footer" :class="{ '-singleColumn': calcSingleColumn }">
```

Comment:
footer has padding. It shouldn't be rendered when it's empty

Reply:
Fixed.